test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -223,9 +224,12 @@ json_spirit::mObject fillBCTest(json_spirit::mObject const& _input)
 	TestBlock genesisBlock(_input.at("genesisBlockHeader").get_obj(), _input.at("pre").get_obj());
 	genesisBlock.setBlockHeader(genesisBlock.blockHeader());
 
-	TestBlockChain testChain(genesisBlock);
+	TestBlockChain::MiningType miningType = getMiningType(_input);
+	TestBlockChain testChain(genesisBlock, miningType);
 	assert(testChain.getInterface().isKnown(genesisBlock.blockHeader().hash(WithSeal)));
 
+	if (_input.count("sealEngine"))
```

Comment:
Is this a new field in blockchain test source files that didn't exist before?

Reply:
yes. but it is not required. we have to support the legacy tests.